streams/src/api/user.rs

```diff
@@ -77,14 +77,17 @@ impl User<()> {
 }
 
 impl<T> User<T> {
-    pub(crate) fn new(user_id: Identity, psks: &[(PskId, Psk)], transport: T) -> Self {
+    pub(crate) fn new<Psks>(user_id: Identity, psks: Psks, transport: T) -> Self
+    where
+        Psks: IntoIterator<Item = (PskId, Psk)>,
+    {
         let mut id_store = KeyStore::new();
         let mut psk_store = HashMap::new();
 
         // Store any pre shared keys
-        for (pskid, psk) in psks {
-            psk_store.insert(*pskid, *psk);
-        }
+        psks.into_iter().for_each(|(pskid, psk)| {
+            psk_store.insert(pskid, psk);
+        });
```

Comment:
You could have left the for loop exactly as it where, why change it?
```suggestion
        for (pskid, psk) in psks {
            psk_store.insert(pskid, psk);
        }
```

Reply:
I assumed if there was an iterative process that using the iterator itself rather than a for loop was more appropriate. From what I saw (https://doc.rust-lang.org/book/ch13-04-performance.html) iterators are more performant so I assumed it was better to use them whenever possible over looping.

Reply:
ok! I doubt it's a meaningful performance improvement in this case, but nvm.